app.py

```diff
@@ -525,7 +525,10 @@ def index(level, step):
 @app.route('/onlinemasters/<level>', methods=['GET'], defaults={'step': 1})
 @app.route('/onlinemasters/<level>/<step>', methods=['GET'])
 def onlinemasters(level, step):
-    g.level = level = int(level)
```

Comment:
I wonder if changing the route to `/onlinemasters/<int:level>/<int:step>` wouldn't do the exact same thing, but in a built-in way?

https://flask.palletsprojects.com/en/2.0.x/api/#url-route-registrations

Reply:
@rix0rrr good idea. Just committed the change you suggested. The behavior is slightly different (we get a generic 404 since the route is outright not matched), but I think it should do.